src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,687 @@ export default class Fms {
+     * @param altitude {number}
+     * @param category {string} one of `FLIGHT_CATEGORY` enum
+     * @param model {AircraftTypeDefinitionModel}
+     * @private
      */
-    setDepartureRunway(runwayModel) {
-        // TODO: this should be an `instanceof` check and should be implemented as part of (or after)
-        // https://github.com/openscope/openscope/issues/93
-        if (!_isObject(runwayModel)) {
-            throw new TypeError(`Expected instance of RunwayModel, but received ${runwayModel}`);
-        }
+    _initializeFlightPlanAltitude(altitude, category, model) {
+        this.flightPlanAltitude = altitude;
 
-        if (this.departureRunwayModel && this.departureRunwayModel.name === runwayModel.name) {
-            return;
+        if (category === FLIGHT_CATEGORY.DEPARTURE) {
+            this.flightPlanAltitude = model.ceiling;
         }
-
-        this.departureRunwayModel = runwayModel;
-
-        this._regenerateSidLeg();
     }
 
     /**
-     * Encapsulates setting of `#arrivalRunwayModel`
+     * Skip ahead to the first waypoint a freshly created FMS should be going to
      *
-     * @for fms
-     * @method setArrivalRunway
-     * @param runwayModel {RunwayModel}
+     * This method is available because we have the ability to spawn aircraft
+     * in "the middle" of any route, rather than requiring them to fly their
+     * route from the beginning. This way, we have the choice to place the aircraft
+     * wherever we want along the route without having to change its contents.
+     *
+     * This method takes an argument that specifies which fix to go to after spawn,
+     * and if not specified, the FMS will target the second fix (because they are)
+     * spawned AT the first fix.
+     *
+     * @for Fms
+     * @method _initializePositionInRouteToBeginAtFixName
+     * @param fixName {string}
+     * @private
      */
-    setArrivalRunway(runwayModel) {
-        // TODO: this should be an `instanceof` check and should be implemented as part of (or after)
-        // https://github.com/openscope/openscope/issues/93
-        if (!_isObject(runwayModel)) {
-            throw new TypeError(`Expected instance of RunwayModel, but received ${runwayModel}`);
+    _initializePositionInRouteToBeginAtFixName(fixName, category) {
+        if (category === FLIGHT_CATEGORY.DEPARTURE) {
+            return;
         }
 
-        if (this.arrivalRunwayModel && this.arrivalRunwayModel.name === runwayModel.name) {
-            return;
+        if (_isNil(fixName)) {
+            return this.moveToNextWaypoint();
         }
 
-        this.arrivalRunwayModel = runwayModel;
+        if (!this._routeModel.hasWaypointName(fixName)) {
+            throw new TypeError(`Expected initial fix to be in flight plan route, but received '${fixName}'`);
+        }
 
-        this._regenerateStarLeg();
+        this.skipToWaypointName(fixName);
     }
 
+    // ------------------------------ PUBLIC ------------------------------
```

Comment:
Public methods should be next after init methods, then private after that.

Reply:
that's what we've got:
- constructor & getters (no label)
- "lifecycle"
- "public"
- "private"

considering done.